cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperatorMockTest.java

```diff
@@ -122,4 +127,19 @@ public void testCreateUpdateDelete(TestContext context) {
             deleteAsync.complete();
         });
     }
+
+    /** Test that the cluster resources will be discovered even when the DELETE event is missed */
+    @Test
+    public void testReconcileAllDeleteCase(TestContext context) throws InterruptedException {
+        KafkaConnectAssemblyOperator kco = createConnectCluster(context);
+        LOGGER.info("Reconciling again -> delete");
+        mockClient.customResources(Crds.kafkaConnect(),
+                KafkaConnectAssembly.class, KafkaConnectAssemblyList.class, DoneableKafkaConnectAssembly.class).
+                inNamespace(NAMESPACE).withName(CLUSTER_NAME).delete();
+        kco.reconcileAll("test-trigger", NAMESPACE, Labels.EMPTY).await(60, TimeUnit.SECONDS);
+
```

Comment:
This is the point where we need to be able to assert that there should be no resources in the mockKube at all. I realise you might not want to do that right now, but could you add a `//TODO``, or open a tech debt issue for that?

Reply:
Done both ... TODO as well as issue.